templates/centos6.tpl.yaml

```diff
@@ -1,4 +1,4 @@
-{% set version =  "0.7.0" %}
```

Comment:
I think we should add new entries to `generate-templates.yaml` instead of completely forgetting about the old templates.

Proposed flow:
1. Copy `centos6.tpl.yaml` to `centos6.tpl.v0.12.0.yaml`
1.1 Repeat 1 for all OSs
2. Modify `v0.12.0` templates to support the new DV flow
3. Add entries for those new templates in `generate-templates.yaml`
4. Reconcile both template versions for the following reasons:
- If a user of CNV 2.4 created a VM with a template, and for some reason, something happened to that template post 2.5 upgrade, the operator can reconcile that template.
- It gives us the flexibility of gracefully deprecating the templates in the future (by removing the
```
os.template.kubevirt.io/{{ osl }}: "true"
workload.template.kubevirt.io/{{ item.workload }}: "true"
flavor.template.kubevirt.io/{{ item.flavor }}: "true"
``` 
labels)

The downside of this is that we will be reconciling a large amount of templates over time, this might be solved if we decide to remove support for old templates for every major CNV version (3.0 for example)
What do you think?

Reply:
I don't like the idea to have version in name of file. I would prefer to name it `centos6-dv.tpl.yaml / centos6-pvc.tpl.yaml`.

Reply:
> I don't like the idea to have version in name of file. I would prefer to name it `centos6-dv.tpl.yaml / centos6-pvc.tpl.yaml`.

On the long run, this is problematic, if the dv templates are modified in the future, what would you call the new ones? dv-2?
Versioning the template files is the only consistent way I could think of that allows us to deprecate templates in the future, and because the DV templates are successors to the PVC templates, they should be a new version of them.
I think that you can also have directories for each version to make it more readable and maintainable, for example:
```
templates/
    | - v0.12.0/
            | - centos6.tpl.yaml
    | - v.0.11.0/
            | - centos6.tpl.yaml
```
and when a release comes, combine all templates from all versions into a single, what do you think?